ynput/ayon-core
# Houdini local render support

## Changelog Description
This PR adds support for local rendering in Houdini and marking render as reviewable.

## Additional Info: How it works 
The logic of this PR is based on re-using the collected data for each render instance to perform local rendering. 
this happens by remapping the collected data `expectedFiles` where I use them to create a dedicated instance for each AOV in the `expectedFiles` while keeping the original instance empty. 
I couldn't remove the original instance while this process because it holds the debug data.

- CollectFarmInstances 
- Collect{..}RenderProducts (mantra, karma, redshift, arnold, vray) 
   Compute `expectedFiles` which is a dictionary of AOVs and their files.
- CollectLocalRenderInstances
   if farm disabled append `render.local.hou` to families
   create instance for each AOV in expected files and add essential data and its representation data
- ExtractLocalRender
   if farm disabled, if `skip_render` disabled, render the node
- Deadline stuff should work only if farm is enabled.


## Testing notes:
1. Open Houdini 
2. Create Render instance  while disabling `Submitting to Farm`
3. Publish (it should work)
4. Create Render instance while enabling `Submitting to Farm`
5. Publish (it should work as usual)
